### DISCUSSION TITLE
AllenaMondo (17/03/2023)

### DISCUSSION THREAD

**Discussion by progettistamatto** [sapienzastudentsnetwork/basi-di-dati-2#34] [Esami]
In questa discussion a tema il progetto "<ins>AllenaMondo</ins>", oggetto del testo di esame del 17 marzo 2023, discuteremo di eventuali proposte di soluzione, dubbi o scelte progettuali in merito alla realizzazione del progetto in questione

**N.B.:** le specifiche e gli use case richiesti dal progetto in questione sono perfettamente equivalenti a quelli richiesti nel progetto oggetto del testo di esame del 10 luglio 2018, dal titolo "Workouts"

**Comment by progettistamatto** [comment]
Per questo esame è disponibile [[QUI]](https://github.com/Exyss/university-notes/raw/main/Bachelor/Secondo%20Anno/Basi%20di%20Dati%20II/AllenaMondo.pdf) una soluzione proposta da Simone Bianco (Exyss)

> [!Warning]
_Le soluzioni di Simone Bianco fanno uso degli **Entity-Relationship (ER)** Diagrams su cui il corso si basava fino all'Anno Accademico 2023/2024, mentre **attualmente vengono utilizzati gli UML Class** Diagram. Sebbene i due siano estremamente simili, vi sono **alcune discordanze tra di loro**, in particolare l'**inversione di posizione dei vincoli di molteplicità agli estremi delle relationship**. Nonostante ciò, il rimanente delle fasi di analisi e progettazione delle soluzioni proposte risulta inaffetto da tale cambiamento. Per tanto, una volta comprese le differenze tra ER e Classi UML, le soluzioni risultano normalmente fruibili._

Rispondi a questo commento taggando @Exyss per eventuali dubbi o segnalazioni in merito

**Comment by progettistamatto** [comment]
Per questo progetto è disponibile [[QUI]](https://github.com/SimoneLid/BD2/raw/main/Prove%20D'Esame/2023-3-17-AllenaMondo/AllenaMondo.pdf) una soluzione proposta da Simone Lidonnici (@SimoneLid)

**Comment by ZhabchenkoNataliya** [comment]
vorrei chiedere cosa significano le lettere blu X, T, R vicino ai vincoli? Grazie

**Reply by Exyss** [reply]
Ciao, scusa la tarda risposta.
Usavo le lettere come un segnalino personale in questo modo:
- X = Vincolo implementato senza Trigger e senza ristrutturazione
- R = Vincolo ristrutturato
- T = Vincolo implementato con Trigger
